recipes/rmlui/4.x/conanfile.py

```diff
@@ -85,78 +86,96 @@ def requirements(self):
-            self._cmake.definitions["ENABLE_TRACY_PROFILING"] = False
-            self._cmake.definitions["MATRIX_ROW_MAJOR"] = self.options.matrix_mode == "row_major"
-            self._cmake.definitions["NO_FONT_INTERFACE_DEFAULT"] = self.options.font_interface is None
-            self._cmake.definitions["NO_THIRDPARTY_CONTAINERS"] = not self.options.with_thirdparty_containers
-
-            self._cmake.configure()
-
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_LUA_BINDINGS"] = self.options.with_lua_bindings
+        tc.cache_variables["BUILD_SAMPLES"] = False
+        tc.cache_variables["DISABLE_RTTI_AND_EXCEPTIONS"] = not self.options.enable_rtti_and_exceptions
+        tc.cache_variables["ENABLE_PRECOMPILED_HEADERS"] = True
+        tc.cache_variables["ENABLE_TRACY_PROFILING"] = False
+        tc.cache_variables["MATRIX_ROW_MAJOR"] = self.options.matrix_mode == "row_major"
+        tc.cache_variables["NO_FONT_INTERFACE_DEFAULT"] = not self.options.font_interface
+        tc.cache_variables["NO_THIRDPARTY_CONTAINERS"] = not self.options.with_thirdparty_containers
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        # The *.cmake files that conan generates using cmake_find_package for CMake's find_package to consume use
-        # different variable naming than described in CMake's documentation, thus the need for most of the replacements.
-        # References:
-        #  * https://cmake.org/cmake/help/latest/module/FindFreetype.html
-        #  * https://cmake.org/cmake/help/latest/module/FindLua.html
+
+        # Since we have switched to CMakeDeps, the targets are named differently, so we need to patch
+        # the CMakeLists.txt to use the new names
         replace_mapping = {
-            "FREETYPE_FOUND": "Freetype_FOUND",
-            "FREETYPE_INCLUDE_DIRS": "Freetype_INCLUDE_DIRS",
-            "FREETYPE_LINK_DIRS": "Freetype_LINK_DIRS",
-            "FREETYPE_LIBRARY": "Freetype_LIBRARIES",
-            "FREETYPE_LIBRARIES": "Freetype_LIBRARIES",
-            "LUA_FOUND": "lua_FOUND",
-            "LUA_INCLUDE_DIR": "lua_INCLUDE_DIR",
-            "LUA_LIBRARIES": "lua_LIBRARIES",
+            "if(FREETYPE_FOUND)": "if(TRUE)", # only in 4.0
+            "include_directories(${FREETYPE_INCLUDE_DIRS})": "",    # only in 4.0
+            "link_directories(${FREETYPE_LINK_DIRS})": "",    # only in 4.0
+            "list(APPEND CORE_LINK_LIBS ${FREETYPE_LIBRARY})": "list(APPEND CORE_LINK_LIBS freetype)", # only in 4.0
+            "list(APPEND CORE_LINK_LIBS ${FREETYPE_LIBRARIES})": "list(APPEND CORE_LINK_LIBS freetype)",
+            "list(APPEND CORE_INCLUDE_DIRS ${FREETYPE_INCLUDE_DIRS})": "",
+            "if(LUA_FOUND)": "if(TRUE)", # only in 4.0
```

Comment:
It would be much better if these were actual patch files since that has added safety and is less likely to have problems in the future

Reply:
I went ahead and made patch files for them